src/proxy/MapProxy.ts

```diff
@@ -552,32 +552,42 @@ export class MapProxy<K, V> extends BaseProxy implements IMap<K, V> {
             }
         };
         var request: ClientMessage;
+        var encodeFunc: Function;
         var handler: Function;
         var responser: Function;
         if (key && predicate) {
             var keyData = this.toData(key);
             var predicateData = this.toData(predicate);
-            request = MapAddEntryListenerToKeyWithPredicateCodec.encodeRequest(this.name, keyData,
-                predicateData, includeValue, flags, false);
+            encodeFunc = (localOnly: boolean) => {
```

Comment:
Doesn't the encodeFunc always create the same message? so why not just use that encoded clientMessage instead and save some cpu cycles ?

Reply:
I think it can be done that way. We actually needed to pass encode function to listenerservice for only de-registration because listener-id that user sees and actual listener-id are different. So I made both registration and de-registration apis similar.